File: app/src/main/java/com/example/delivery/MainActivity.java
package com.example.delivery;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;
import androidx.appcompat.widget.ActionMenuView;
import androidx.databinding.DataBindingUtil;
import androidx.lifecycle.ViewModelProvider;
import androidx.navigation.NavController;
import androidx.navigation.fragment.NavHostFragment;

import android.os.Bundle;
import android.view.Menu;
import android.view.MenuItem;

import com.example.delivery.menu.adapters.ImageBindingAdapter;
import com.example.delivery.databinding.ActivityMainBinding;
import com.example.delivery.mapspoint.MapspointViewModel;
import com.example.delivery.menu.MainViewModel;

public class MainActivity extends AppCompatActivity {
    private ActivityMainBinding binding;
    private NavHostFragment navHostFragment;
    private NavController navController;
    private Permission permission;
    private ImageBindingAdapter imageBindingAdapter;

    @Override
    public void onRequestPermissionsResult(int requestCode, @NonNull String[] permissions, @NonNull int[] grantResults) {
        permission.permissionsResult(requestCode,grantResults);
    }

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        binding=DataBindingUtil.setContentView(this,R.layout.activity_main);

        navHostFragment=(NavHostFragment)getSupportFragmentManager().findFragmentById(R.id.nav_host_fragment);
        navController=navHostFragment.getNavController();
        setSupportActionBar(binding.toolbarUp);
        getSupportActionBar().setDisplayShowTitleEnabled(true);

        //====================================ViewModels==================================
        MapspointViewModel mapspointViewModel=new ViewModelProvider(this).get(MapspointViewModel.class);
        MainViewModel menuViewModel=new ViewModelProvider(this).get(MainViewModel.class);
        menuViewModel.getAllData();
        //====================================Permissions==================================
        permission= new Permission(mapspointViewModel, this, this);
        /* ...and GeoPoint  */
        permission.requestPerm();
    }

    @Override
    public boolean onOptionsItemSelected(@NonNull MenuItem item) {
        switch (item.getItemId()){
            case R.id.cart:
                Loger.log("cart is clicked");
                break;
            case R.id.menu:
                navController.navigate(R.id.fragmentMenuList);
                Loger.log("menu is clicked");
                break;
            case R.id.favorites:
                navController.navigate(R.id.fragmentFavorites);
                Loger.log("favorites is clicked");
                break;
            case R.id.orders:
                navController.navigate(R.id.fragmentOrders);
                Loger.log("orders is clicked");
                break;
            case R.id.restaurants:
                navController.navigate(R.id.fragmentMapsPoint);
                Loger.log("restaurants is clicked");
                break;
            case R.id.profile:
                navController.navigate(R.id.fragmentProfile);
                Loger.log("profile is clicked");
                break;
        }
        return super.onOptionsItemSelected(item);
    }

    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        // Inflate the toolbar menu
        getMenuInflater().inflate(R.menu.toolbar_menu_up, menu);

        // Inflate and initialize the bottom menu
        ActionMenuView bottomBar = binding.toolbarDown;
        Menu bottomMenu = bottomBar.getMenu();
        getMenuInflater().inflate(R.menu.toolbar_menu_down, bottomMenu);
        for (int i = 0; i < bottomMenu.size(); i++) {
            bottomMenu.getItem(i).setOnMenuItemClickListener(new MenuItem.OnMenuItemClickListener() {
                @Override
                public boolean onMenuItemClick(MenuItem item) {
                    return onOptionsItemSelected(item);
                }
            });
        }
        return true;
    }
}

File: app/src/main/java/com/example/deliverykotlin/MainActivity.kt
package com.example.deliverykotlin

import android.os.Bundle
import android.view.Menu
import android.view.MenuItem
import android.view.View
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity
import androidx.databinding.DataBindingUtil.setContentView
import androidx.navigation.NavController
import androidx.navigation.Navigation
import androidx.navigation.ui.AppBarConfiguration
import androidx.navigation.ui.NavigationUI
import com.example.deliverykotlin.components.menu.MenuViewModel
import com.example.deliverykotlin.databinding.ActivityMainBinding
import org.koin.android.viewmodel.ext.android.viewModel


class MainActivity : AppCompatActivity() {
    private val viewModel : MenuViewModel by viewModel()
    private var textCartItemCount: TextView? = null
    var itemCart:MenuItem?=null
    lateinit var navController: NavController

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        textCartItemCount=findViewById(R.id.textCartItemCount)

        val binding=setContentView<ActivityMainBinding>(this, R.layout.activity_main)
        setSupportActionBar(findViewById(R.id.toolbar))

        val appBarConfiguration = AppBarConfiguration.Builder(
                R.id.navigation_menu, R.id.navigation_favorites, R.id.navigation_orders, R.id.navigation_points, R.id.navigation_profile
        ).build()

        val navView = binding.navView
        navController = Navigation.findNavController(this, R.id.nav_host_fragment)
        NavigationUI.setupActionBarWithNavController(this, navController, appBarConfiguration)
        NavigationUI.setupWithNavController(navView, navController)

        viewModel.getMenuList()
    }

    override fun onCreateOptionsMenu(menu: Menu): Boolean {
        menuInflater.inflate(R.menu.menu_main, menu)

        val item: MenuItem = menu.findItem(R.id.cart)

        itemCart=item
        //itemCart!!.setVisible(false)
        return true
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        // Handle action bar item clicks here. The action bar will
        // automatically handle clicks on the Home/Up button, so long
        // as you specify a parent activity in AndroidManifest.xml.
        return when (item.itemId) {
            R.id.action_settings -> true
            android.R.id.home -> {
                navController.popBackStack(); true
            }
            else -> super.onOptionsItemSelected(item)
        }
    }
}